fix(docs-infra): add cookie consent gtag event default state

bencodezen opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
No Google Analytics consent code existed.

Issue Number: N/A


## What is the new behavior?

Google Analytics code for cookie consent is now added per the [docs](https://developers.google.com/tag-platform/security/guides/consent?consentmode=basic#gtag.js_4).

The other half of this PR that grants cookie consent can be found at https://github.com/angular/dev-infra/pull/1805

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

AndrewKushnir commented on aio/src/app/shared/analytics.service.ts:
```diff
@@ -90,6 +90,17 @@ export class AnalyticsService {
     window.gtag = function() { window.dataLayer?.push(arguments); };
+
+    // Cookie banner consent initial state
+    window.gtag('content', 'default', {
```
@bencodezen could you please provide some additional info (may be add a comment to the code) on where/when we change values of those fields (I don't see it in the code)?

AndrewKushnir commented:
@bencodezen could you please also change the commit message to start with `fix(docs-infra): ...`, so that we can merge the change into the patch branch (feature commits can only go to the minor/major branches)?

bencodezen commented:
> @bencodezen could you please also change the commit message to start with `fix(docs-infra): ...`, so that we can merge the change into the patch branch (feature commits can only go to the minor/major branches)?

@AndrewKushnir Absolutely! Thanks for calling that out. I've updated the PR with the correct doc comment and message!

bencodezen commented on adev/src/app/core/services/analytics/analytics.service.ts:
```diff
@@ -64,6 +64,19 @@ export class AnalyticsService {
+      ad_storage: 'denied',
+      analytics_storage: 'denied',
+      personalization_storage: 'denied',
+      wait_for_update: 500
```
@AndrewKushnir Thanks for the feedback!

Thanks for catching the discrepancies. I'm actually not sure how the two got misaligned in terms of properties, but you're totally right in that it would be best to refactor this sooner rather than later to avoid conflicts like this. 

Let me update the PR and see what I can do. If the refactoring proves to be too complicated for this PR, I'll make sure to note it so we can land this and then push the PR refactor afterwards!

AndrewKushnir replied:
@bencodezen I still see a different set of props here and in https://github.com/angular/dev-infra/pull/1805/files (1 extra in this PR). Could you please let me know if this is expected?

bencodezen replied:
@AndrewKushnir This is correct. The delay on the initial consent is there to allow async operations to occur if necessary.

AndrewKushnir commented:
@bencodezen LGTM, thanks for the updates. I've removed the "review" label, please add the "merge" label when the PR is ready for merge. Thank you.

atscott commented:
@bencodezen It looks like the CI failure is legitimate. Could you take a look?

bencodezen commented:
Thanks for the ping @atscott! I'm in the midst of trying to figure out why the export isn't found by the CI tool. Will let you know when all the tests are passing as expected.

devversion reviewed (approved):
LGTM for dev-infra (been waiting since this is blocked, but not ready for review— but want to get it off my list)

bencodezen committed:
fix(docs-infra): add cookie consent gtag event default state

This PR sets a default state for cookie consent of 'denied'. The other part of the PR exists in the angular/dev-infra repo which will grant permission when the user accepts the cookie banner.

bencodezen commented:
Finally managed to get everything passing in the CI! 🥳 Ready for review and then merge!

AndrewKushnir commented:
This PR was merged into the repository by commit 0b53fdb3b41f2517bcfebee75295aa62769b3e04.
